env_docker.go

```diff
@@ -131,8 +132,12 @@ func New(opts ...EnvironmentOption) (_ *DockerEnvironment, err error) {
 
 	switch runtime.GOOS {
 	case "darwin":
-		d.hostAddr = dockerMacOSGatewayAddr
+		d.hostAddr = dockerGatewayAddr
 	default:
+		if d.WSL2() {
```

Comment:
What does runtime.GOOS return if running on WSL? If it isn't linux what is it? 

This switch statement is also redundant, you really shouldn't have to branch conditionals inside of a `switch`. This should either be in a `case` or should not be set here. 

I also know it was like this before so 🤷🏽

Reply:
> What does runtime.GOOS return if running on WSL? If it isn't linux what is it?

It does return `linux`, that's why the implementation checks for the kernel version by reading from `/proc/version`. Also it uses `runtime.GOOS != linux` as a condition to fail quickly returning `false`.